src/static/riot/analytics/analytics.tag

```diff
@@ -0,0 +1,425 @@
+            <div class="label">
+                Users Joined
+            </div>
+        </div>
+
+        <div class='chart-container'>
+            <canvas ref="user_chart"></canvas>
+        </div>
+    </div>
+
+    <a class="ui green button" href="{ URLS.ANALYTICS_API(start_date_string, end_date_string, time_unit, 'csv') }" download="codalab_analytics.csv">
+        <i class="icon download"></i>Download as CSV
+    </a>
+
+    <script>
+        var self = this
+
+        /*---------------------------------------------------------------------
+         Init
+        ---------------------------------------------------------------------*/
+        self.errors = []
+
+        self.competitionsChart;
+        self.submissionsChart;
+        self.usersChart;
+        self.time_unit = 'month';
+
+        self.competitions_data;
+        self.submissions_data;
+        self.users_data;
+
+        let datetime = luxon.DateTime
+        self.start_date = datetime.local(datetime.local().year)
+        self.end_date = datetime.local()
+
+        self.one("mount", function () {
+            // Semantic UI
+            $('.tabular.menu .item', self.root).tab();
+
+            self.shortcut_dropdown = $(self.refs.date_shortcut_dropdown)
+            self.resolution_dropdown = $(self.refs.chart_resolution_dropdown)
+            self.shortcut_dropdown.dropdown({
+                onChange: function(value, text, item) {
+                    if (value === 'custom') {
+                        $(self.refs.date_selection_container).removeClass('hidden')
+                        self.resolution_dropdown.dropdown('set selected', 'day')
+                    } else {
+                        $(self.refs.date_selection_container).addClass('hidden')
+                        self.time_range_shortcut(value)
+                    }
+                }
+            })
+
+            self.resolution_dropdown.dropdown({
+                onChange: function(value, text, item) {
+                    self.update_chart_resolution(value)
+                }
+            })
+
+            /*---------------------------------------------------------------------
+             Calendar Setup
+            ---------------------------------------------------------------------*/
+
+            let general_calendar_options = {
+                type: 'date',
+                // Sets the format of the placeholder date string to YYYY-MM-DD
+                formatter: {
+                   date: function (date, settings) {
+                       return datetime.fromJSDate(date).toISODate()
+                   }
+                },
+            }
+
+            let start_specific_options = {
+                endCalendar: $(self.refs.end_calendar),
+                onChange: function(date, text) {
+                    self.start_date = datetime.fromJSDate(date)
+
+                    let end_date = $(self.refs.end_calendar).calendar('get date')
+                    if (end_date != null) {
```

Comment:
If i'm tracking w/ this logic, you are checking if you have an end date set at all. This can be done w/
```javascript
if (!!end_date) {

}
```

Reply:
Done.